# Trabajo Práctico Integrador - Introducción al Análisis de Datos

**Caso:** análisis y predicción de cancelaciones en reservas hoteleras.

## 1. Presentación breve del problema

Una empresa hotelera desea estudiar el comportamiento de sus reservas para identificar patrones asociados a la cancelación de estadías. Para ello, dispone de información sobre reservas realizadas en dos tipos de establecimientos: hoteles urbanos y hoteles tipo resort.

El dataset incluye información sobre fechas de llegada, duración de la estadía, cantidad de huéspedes, canal de reserva, tipo de cliente, historial de cancelaciones, tarifa promedio por noche y estado final de la reserva.

**Variable objetivo:** `is_canceled`

- `0`: reserva no cancelada.
- `1`: reserva cancelada.

##2. Datos de identificación

- **Apellido y nombre:** Rodriguez Joaquin , Pichulman Miguel
- **Comisión:** 12
- **Dataset asignado:** DATASET L
- **Entrega:** 1
- **Fecha:**

##3. Carga de Datos

##3.1 Importación de librerías

In [ ]:
# Librerías básicas
import pandas as pd

## 3.2 Carga del dataset

Cargar del archivo CSV correspondiente a la comisión numero 12, Dataset L.

In [ ]:
from google.colab import files
uploaded = files.upload()

df = pd.read_csv("hotel booking TPI grupo L.csv")

## 4. Vista Inicial

Visualizacion de las primeras filas del Dataset.

In [ ]:
df.head()

## 5. Estructura general

Cantidad de filas, cantidad de columnas y nombres de las variables

In [ ]:
filas, columnas = df.shape
print("Filas: ", filas)
print("\nColumnas: ", columnas)

print("Variables")
print(df.columns.tolist())


Filas:  25000

Columnas:  32
Variables
['booking_id', 'hotel', 'is_canceled', 'lead_time', 'arrival_date_year', 'arrival_date_month', 'arrival_date_week_number', 'arrival_date_day_of_month', 'arrival_date', 'stays_in_weekend_nights', 'stays_in_week_nights', 'adults', 'children', 'babies', 'meal', 'country', 'market_segment', 'distribution_channel', 'is_repeated_guest', 'previous_cancellations', 'previous_bookings_not_canceled', 'reserved_room_type', 'assigned_room_type', 'booking_changes', 'deposit_type', 'agent', 'company', 'days_in_waiting_list', 'customer_type', 'adr', 'required_car_parking_spaces', 'total_of_special_requests']


### 6 Tipos de variables

  Clasificacion inicial de las variables numericas, categoricas y temporales

In [ ]:
# Código de análisis
print("Tipos de datos")
print(df.dtypes)

Tipos de datos
booking_id                         object
hotel                              object
is_canceled                         int64
lead_time                           int64
arrival_date_year                   int64
arrival_date_month                 object
arrival_date_week_number            int64
arrival_date_day_of_month           int64
arrival_date                       object
stays_in_weekend_nights             int64
stays_in_week_nights                int64
adults                              int64
children                          float64
babies                              int64
meal                               object
country                            object
market_segment                     object
distribution_channel               object
is_repeated_guest                   int64
previous_cancellations              int64
previous_bookings_not_canceled      int64
reserved_room_type                 object
assigned_room_type                 object
booking_changes    

## 7. Caracterizacion descriptiva inicial

Exploracion basica de las variables mediante resumenes que permitan conocer sus valores, rangos o comportamientos generales, acompañados por observaciones escritas.

In [ ]:
df.describe().round(2)


,is_canceled,lead_time,arrival_date_year,arrival_date_week_number,arrival_date_day_of_month,stays_in_weekend_nights,stays_in_week_nights,adults,children,babies,is_repeated_guest,previous_cancellations,previous_bookings_not_canceled,booking_changes,agent,company,days_in_waiting_list,adr,required_car_parking_spaces,total_of_special_requests
count,25000.00,25000.00,25000.00,25000.00,25000.00,25000.00,25000.00,25000.00,24999.0,25000.00,25000.00,25000.00,25000.00,25000.00,21578.00,1385.00,25000.00,25000.00,25000.00,25000.00
mean,0.37,104.10,2024.16,26.65,15.76,0.93,2.51,1.85,0.1,0.01,0.03,0.09,0.13,0.22,87.08,185.96,2.30,102.40,0.06,0.56
std,0.48,106.57,0.71,13.34,8.80,1.01,1.93,0.51,0.4,0.09,0.17,0.85,1.50,0.65,110.94,130.03,17.46,58.33,0.24,0.79
min,0.00,0.00,2023.00,1.00,1.00,0.00,0.00,0.00,0.0,0.00,0.00,0.00,0.00,0.00,1.00,9.00,0.00,-6.38,0.00,0.00
25%,0.00,18.00,2024.00,16.00,8.00,0.00,1.00,2.00,0.0,0.00,0.00,0.00,0.00,0.00,9.00,62.00,0.00,70.00,0.00,0.00
50%,0.00,70.00,2024.00,27.00,16.00,1.00,2.00,2.00,0.0,0.00,0.00,0.00,0.00,0.00,14.00,174.00,0.00,95.00,0.00,0.00
75%,1.00,161.00,2025.00,37.00,23.00,2.00,3.00,2.00,0.0,0.00,0.00,0.00,0.00,0.00,229.00,255.00,0.00,126.00,0.00,1.00
max,1.00,629.00,2025.00,52.00,31.00,16.00,40.00,20.00,3.0,2.00,1.00,26.00,70.00,21.00,531.00,543.00,391.00,5400.00,2.00,5.00


---

### 8. Primeras observaciones

Comentarios generales sobre la informacion descubierta hasta el momento.



*   En un primer analisis de la variable objetivo <b>is_canceled</b>, podemos decir que existe una proporcion considerable de reservas canceladas.
*   Los tiempos de reserva (<b>lead_time</b>) muestran un rango muy amplio, con valores minimos de 0 días hasta varios 629 dias. La desviacion estandar sera un punto de interes.
* Se detectaron variables con muchos valores nulos, apuntando directamente a <b>company</b> y <b>agent</b>.
* Se observa valores atipicos (outliers) en variables clave que requeriran tratamiento, como tarifas diarias <b>(adr)</b> con montos negativos o registros que indican hasta 20 adultos en una sola reserva.

##9. Consideraciones eticas iniciales
Reflexion breve sobre el uso responsable de los datos y los limites de las conclusiones que pueden obtenerse en la primera etapa.

Al trabajar con datos del sector hotelero, la prioridad debe ser proteger la privacidad de los huéspedes. Aunque la identidad en este dataset se mantiene en el anonimato y no contiene información de identificación directa, variables como la nacionalidad (country) o las peticiones especiales deben analizarse de manera objetiva, evitando interpretaciones que puedan llevar a una clasificación inadecuada de los huéspedes. Asimismo, es importante establecer los límites de esta etapa: dado que los datos aún presentan ruido, valores nulos y valores atípicos, no resulta ético ni profesional utilizarlos para fundamentar decisiones de negocio definitivas sin realizar previamente un proceso de limpieza, validación y control de calidad.